webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/UserSessionInvalidationFilter.scala

```diff
@@ -132,3 +132,86 @@ class UserSessionInvalidationFilter(userRepository: UserRepository, userDetailLi
+      user match {
+        case None                                => Some("user access to Rudder has been updated")
+        case Some((_, HASH_USER_INVALID_STATUS)) => Some("user status has been updated to an invalid one")
+        case Some(_)                             => None
+      }
+    }
+  }
+
+  // special hash values are negative
+  val HASH_USER_ADMIN:          Int = -1
+  val HASH_USER_INVALID_STATUS: Int = -2
+
+  // ensure the hash is positive, to be able to use special negative values
+  private def simpleHash(username: String, password: String, authz: Set[String]): Int = {
+    (username, password, authz).hashCode() & 0x7fffffff
+  }
+
+  // to this method should be only used to hash active users
+  def hashRudderUserDetail(user: RudderUserDetail): (String, Int) = {
+    if (user.isAdmin) {
+      user.getUsername -> HASH_USER_ADMIN
+    } else {
+      user.getUsername -> simpleHash(user.getUsername, user.getPassword, user.authz.authorizationTypes.map(_.id))
+    }
+  }
+
+  // special case of user that is not logged-in : hashed with known values
+  private[this] def hashRudderUserDetail(user: RudderUserDetail, status: UserStatus): (String, Int) = {
+    if (status.in(UserStatus.Disabled, UserStatus.Deleted)) {
+      user.getUsername -> HASH_USER_INVALID_STATUS
+    } else {
+      hashRudderUserDetail(user)
+    }
+  }
+
+  /**
+    * User status have to be updated from the database, since the loaded RudderUserDetail does not have
+    * the definitive status of the user after a file reload.
+    * This also adds new users to the Ref, having knowledge of their latest hash
+    */
+  def updateUsers(
+      ref:             Array[(String, Int)],
+      newUserStatuses: Map[String, UserStatus]
+  ): Map[String, RudderUserDetail] => IOResult[Unit] = (users) => {
+    val validatedUsers = ref.flatMap {
+      case (id, loggedUser) =>
+        (users.get(id), newUserStatuses.get(id)) match {
+          case (None, _)            => None // user still in Ref but not in new users list should be removed
+          case (Some(user), status) =>
+            // user without status in database is a deleted user
+            Some(hashRudderUserDetail(user, status.getOrElse(UserStatus.Deleted)))
+        }
+    }
+    // file users not yet in ref should be added to ref
+    val fileUsers      = users.flatMap {
+      case (id, user) =>
+        if (!ref.exists(_._1 == id)) {
+          Some(hashRudderUserDetail(user, newUserStatuses.getOrElse(id, UserStatus.Deleted)))
+        } else {
+          None
+        }
+    }
+    // we just need to have another array that would be copied to the final target
+    val sourceArray    = (fileUsers ++ validatedUsers).toArray
+    IOResult.attempt {
+      Array.copy(sourceArray, 0, ref, 0, sourceArray.length)
+    }.chainError(
+      s"User sessions cache may have been corrupted, having more than ${USERS_REF_SIZE} users in Rudder is not supported"
```

Comment:
We never want that kind of case, the cost of having someone out reaching it is too high.
if really it's important, at least you need to use a file/jvm based property that can be changed without a recompilation. The user need to have the agency without you to workaround the limit.